src/plugins/BackgroundFetch/BackgroundFetchPlugin.cs

```diff
@@ -71,90 +73,95 @@ private void RecreateObservable()
             Validates.NotNull(_currentGitUiCommands);
 
             IGitModule gitModule = _currentGitUiCommands.Module;
-            if (fetchInterval > 0 && gitModule.IsValidGitWorkingDir())
+            if (fetchInterval <= 0 || !gitModule.IsValidGitWorkingDir())
             {
-                _cancellationToken =
-                    Observable.Timer(TimeSpan.FromSeconds(Math.Max(5, fetchInterval)))
-                              .SelectMany(i =>
-                              {
-                                  // if git not running - start fetch immediately
-                                  if (!gitModule.IsRunningGitProcess())
-                                  {
-                                      return Observable.Return(i);
-                                  }
+                return;
+            }
+
+            TimeSpan intervalTimeSpan = TimeSpan.FromSeconds(Math.Max(5, fetchInterval));
+            TimeSpan startTimer = _fetchImmediatelyOnRepoOpening.ValueOrDefault(Settings) ? TimeSpan.FromMilliseconds(10) : intervalTimeSpan;
 
-                                  // in other case - every 5 seconds check if git still running
-                                  return Observable
-                                      .Interval(TimeSpan.FromSeconds(5))
-                                      .SkipWhile(ii => gitModule.IsRunningGitProcess())
-                                      .FirstAsync()
-                                  ;
-                              })
-                              .Repeat()
-                              .ObserveOn(ThreadPoolScheduler.Instance)
-                              .Subscribe(i =>
+            _cancellationToken =
+                Observable.Timer(startTimer, intervalTimeSpan)
+                          .SelectMany(i =>
```

Comment:
❔ As you are changing the indent anyway...
```suggestion
            _cancellationToken = Observable
                .Timer(startTimer, intervalTimeSpan)
                .SelectMany(i =>
```

Reply:
Done.